src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -443,6 +444,20 @@ object PruneDeadFields {
         val dep = requestedType.copy(rvRowType = unify(child.typ.rvRowType,
           requestedType.rvRowType.insert(fieldDep, path.toList)._1.asInstanceOf[TStruct]))
         memoizeMatrixIR(child, dep, memo)
+      case MatrixExplodeCols(child, path) =>
+        def getExplodedField(typ: MatrixType): Type = typ.colType.queryTyped(path.toList)._1
+        val baseType = getExplodedField(child.typ)
+        val fieldDep = Try(getExplodedField(requestedType)) match {
```

Comment:
I see this is used in a lot of places already, but, IMO, this is a dangerous pattern because it masks any exception that happened in `getExplodedField(requestedType)`, even exceptions that are not expected. Ideally we would have a method that returns an option. If we must rely on exceptions to fail, we should find out how to make Try only stop one exception and use a brand new exception used only by `queryTypes` to signify that there was a failure to find the field.

Reply:
I switched to try/catch with the specific exception thrown in this case

Reply:
I switched to try/catch with the specific exception thrown in this case